### DISCUSSION TITLE
Override file download from NextJS

### DISCUSSION THREAD

**Discussion by petrosmm** [vercel/next.js#53583] [Help]
### Summary

Hello,

I would like to know if there is a way to override the default functionality of the pipeline that serves files as a download to the user from the /public folder -- to replace a file before letting the user receive it. For example, I may have a (templated) pdf file that I would I would generate some text into  and then pass it as a download. Is middleware the only way to accomplish this? Thoughts?

**Comment by Nadeera3784** [comment]
Hi @petrosmm,
If this is related to generate PDF you might be able to accomplish this by using  [pdf-lib](https://www.npmjs.com/package/pdf-lib).
It will generate PDF on the fly.

**Comment by petrosmm** [comment]
that's fine but I am referring to the situation where I wish to override any file (whether it is a text file or a pdf) and send another file instead? not necessarily editing the PDF itself but how to override the file in the nextjs framework... do we use middleware?
